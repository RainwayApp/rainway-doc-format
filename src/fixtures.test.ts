import { fs } from "memfs";
import { format, Mode } from "./index";

jest.mock("fs", () => {
  return jest.requireActual("memfs");
});

const realFs = jest.requireActual("fs") as typeof fs;

describe("fixtures [api-extractor]", () => {
  beforeAll(async () => {
    const fixturePath = "src/__fixtures__/api-extractor/";
    await fs.promises.mkdir(process.cwd(), { recursive: true });
    const fixtures = await realFs.promises.readdir(fixturePath);
    await Promise.all(
      fixtures.map(async (path) => {
        const src = await realFs.promises.readFile(fixturePath + path, {
          encoding: "utf-8",
        });
        await fs.promises.writeFile(path, src, { encoding: "utf-8" });
      })
    );
  });

  afterAll(async () => {
    await fs.promises.rm(process.cwd(), { recursive: true });
  });

  it("should run for api-extractor", async () => {
    await expect(
      format({
        input: "*.md\n!index.md",
        output: "docs/",
        srcLib: "test-lib",
        dstLib: "your-lib",
        mode: Mode.ApiExtractor,
        home: "https://home.com",
        index: "./your-lib",
      })
    ).resolves.toBe(0);

    expect(fs.existsSync("./docs/your-lib.md")).toBeTruthy();
    expect(fs.existsSync("./docs/your-lib-beginwork.md")).toBeTruthy();
    expect(fs.existsSync("./docs/your-lib-endwork.md")).toBeTruthy();
    await expect(
      fs.promises.readFile("./docs/your-lib.md", { encoding: "utf-8" })
    ).resolves.toMatchSnapshot();
    await expect(
      fs.promises.readFile("./docs/your-lib-beginwork.md", {
        encoding: "utf-8",
      })
    ).resolves.toMatchSnapshot();
    await expect(
      fs.promises.readFile("./docs/your-lib-endwork.md", { encoding: "utf-8" })
    ).resolves.toMatchSnapshot();
  });
});

describe("fixtures [xmldoc]", () => {
  beforeAll(async () => {
    const fixturePath = "src/__fixtures__/xmldoc/";
    await fs.promises.mkdir(process.cwd(), {
      recursive: true,
    });
    const fixtures = await realFs.promises.readdir(fixturePath);
    await Promise.all(
      fixtures.map(async (path) => {
        const src = await realFs.promises.readFile(fixturePath + path, {
          encoding: "utf-8",
        });
        await fs.promises.writeFile(path, src, { encoding: "utf-8" });
      })
    );
  });

  afterAll(async () => {
    await fs.promises.rm(process.cwd(), { recursive: true });
  });

  it("should run for xmldoc", async () => {
    await expect(
      format({
        input: "*.md\n!index.md",
        output: "docs/",
        srcLib: "test-lib",
        dstLib: "your-lib",
        mode: Mode.Xmldoc,
        home: "https://home.com",
        index: "./your-lib",
      })
    ).resolves.toBe(0);

    expect(fs.existsSync("./docs/your-lib.md")).toBeTruthy();
    expect(fs.existsSync("./docs/your-lib-program.md")).toBeTruthy();
    expect(fs.existsSync("./docs/your-lib-program-beginwork.md")).toBeTruthy();
    expect(fs.existsSync("./docs/your-lib-program-endwork.md")).toBeTruthy();
    expect(fs.existsSync("./docs/your-lib-program-main.md")).toBeTruthy();
    expect(fs.existsSync("./docs/your-lib-program-program.md")).toBeTruthy();
    await expect(
      fs.promises.readFile("./docs/your-lib.md", { encoding: "utf-8" })
    ).resolves.toMatchSnapshot();
    await expect(
      fs.promises.readFile("./docs/your-lib-program.md", { encoding: "utf-8" })
    ).resolves.toMatchSnapshot();
    await expect(
      fs.promises.readFile("./docs/your-lib-program-beginwork.md", {
        encoding: "utf-8",
      })
    ).resolves.toMatchSnapshot();
    await expect(
      fs.promises.readFile("./docs/your-lib-program-endwork.md", {
        encoding: "utf-8",
      })
    ).resolves.toMatchSnapshot();
    await expect(
      fs.promises.readFile("./docs/your-lib-program-main.md", {
        encoding: "utf-8",
      })
    ).resolves.toMatchSnapshot();
    await expect(
      fs.promises.readFile("./docs/your-lib-program-program.md", {
        encoding: "utf-8",
      })
    ).resolves.toMatchSnapshot();
  });
});
