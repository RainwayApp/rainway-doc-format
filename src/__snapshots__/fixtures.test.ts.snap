// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fixtures [api-extractor] should run for api-extractor 1`] = `
"<!-- Do not edit this file. It is automatically generated by API Documenter. -->

[Home](https://home.com) &gt; [test-lib](./your-lib)

## test-lib package

## Functions

|  Function | Description |
|  --- | --- |
|  [beginWork()](./your-lib-beginwork) | The function that begins work |
|  [endWork()](./your-lib-endwork) | The function that ends work |

"
`;

exports[`fixtures [api-extractor] should run for api-extractor 2`] = `
"<!-- Do not edit this file. It is automatically generated by API Documenter. -->

[Home](https://home.com) &gt; [test-lib](./your-lib) &gt; [beginWork](./your-lib-beginwork)

## beginWork() function

The function that begins work

<b>Signature:</b>

\`\`\`typescript
export declare function beginWork(): void;
\`\`\`
<b>Returns:</b>

void

"
`;

exports[`fixtures [api-extractor] should run for api-extractor 3`] = `
"<!-- Do not edit this file. It is automatically generated by API Documenter. -->

[Home](https://home.com) &gt; [test-lib](./your-lib) &gt; [endWork](./your-lib-endwork)

## endWork() function

The function that ends work

<b>Signature:</b>

\`\`\`typescript
export declare function endWork(): void;
\`\`\`
<b>Returns:</b>

void

"
`;

exports[`fixtures [xmldoc] should run for xmldoc 1`] = `
"# test-lib assembly

## test_lib namespace

| public type | description |
| --- | --- |
| class [Program](./your-lib-program) | Program entrypoint |

<!-- DO NOT EDIT: generated by xmldocmd for test-lib.dll -->
"
`;

exports[`fixtures [xmldoc] should run for xmldoc 2`] = `
"# Program class

Program entrypoint

\`\`\`csharp
public class Program
\`\`\`

## Public Members

| name | description |
| --- | --- |
| [Program](./your-lib-program-program)() | The default constructor. |
| static [BeginWork](./your-lib-program-beginwork)() | Begins the work |
| static [EndWork](./your-lib-program-endwork)() | Ends the work |
| static [Main](./your-lib-program-main)(…) | The entrypoint |

## See Also

* namespace [test_lib](./your-lib)

<!-- DO NOT EDIT: generated by xmldocmd for test-lib.dll -->
"
`;

exports[`fixtures [xmldoc] should run for xmldoc 3`] = `
"# Program.BeginWork method

Begins the work

\`\`\`csharp
public static void BeginWork()
\`\`\`

## See Also

* class [Program](./your-lib-program)
* namespace [test_lib](./your-lib)

<!-- DO NOT EDIT: generated by xmldocmd for test-lib.dll -->
"
`;

exports[`fixtures [xmldoc] should run for xmldoc 4`] = `
"# Program.EndWork method

Ends the work

\`\`\`csharp
public static void EndWork()
\`\`\`

## See Also

* class [Program](./your-lib-program)
* namespace [test_lib](./your-lib)

<!-- DO NOT EDIT: generated by xmldocmd for test-lib.dll -->
"
`;

exports[`fixtures [xmldoc] should run for xmldoc 5`] = `
"# Program.Main method

The entrypoint

\`\`\`csharp
public static void Main(string[] args)
\`\`\`

| parameter | description |
| --- | --- |
| args | arguments |

## See Also

* class [Program](./your-lib-program)
* namespace [test_lib](./your-lib)

<!-- DO NOT EDIT: generated by xmldocmd for test-lib.dll -->
"
`;

exports[`fixtures [xmldoc] should run for xmldoc 6`] = `
"# Program constructor

The default constructor.

\`\`\`csharp
public Program()
\`\`\`

## See Also

* class [Program](./your-lib-program)
* namespace [test_lib](./your-lib)

<!-- DO NOT EDIT: generated by xmldocmd for test-lib.dll -->
"
`;
